File: public/src/core/models/dto/response/IDoctorService.ts
import { CreateDoctorResponse } from "../models/dto/response/CreateDoctorResponse";
import { FindAllDoctorResponse } from "../models/dto/response/FindAllDoctorResponse";

export interface IDoctorService {
  findAll(): Promise<Array<FindAllDoctorResponse>>;

  create(doctor: CreateDoctorRequest): Promise<CreateDoctorResponse>;
}

File: api/src/main/kotlin/com/study/api/core/service/IDoctorService.kt
package com.study.api.core.service

import com.study.api.core.models.Doctor
import com.study.api.core.models.dto.requests.CreateDoctorRequest
import com.study.api.core.models.dto.requests.UpdateDoctorRequest
import com.study.api.core.models.dto.responses.*

interface IDoctorService {

    fun findAll(page: Int, size: Int): List<FindAllDoctorResponse>

    fun findById(id: Long): FindByIdDoctorReponse?

    fun create(doctor: CreateDoctorRequest): CreateDoctorResponse

    fun delete(id: Long): DeleteDoctorResponse

    fun update(doctor: UpdateDoctorRequest): UpdateDoctorResponse

}